examples/fpga/artya7/util/vivado_hook_route_design_post.tcl

```diff
@@ -0,0 +1,34 @@
+open_project ../../lowrisc_ibex_top_artya7_0.1.xpr
+set saif_name "detailed_power.saif"
+
+# TODO: All cells are removed in optimization process. Needs investigation, removal happens because
+# of unloaded cells. Until it is solved, this workaround generates another implementation run where
+# optimization is skipped. That way functional post-implementation simulation can be done. create_run
+# command requires flow version, which needs to be extracted.
+regexp -- {Vivado v([0-9]{4})\.[0-9]} [version] -> year
+create_run impl_2 -parent_run synth_1 -flow "Vivado Implementation $year"
+set_property STEPS.OPT_DESIGN.IS_ENABLED false [get_runs impl_2]
+launch_runs impl_2
+wait_on_run impl_2
+open_run impl_2
+
+# Runs a post implementation functional simulation with the memory initialized with SRAMInitFile.
+# Feeds clock (100mhz) and reset switch and records switching activity for 3ms.
+if {![file exists "$saif_name"]} {
```

Comment:
I'm not sure this is the right thing to do: everything else gets rebuilt from scratch on a fusesoc run. Probably we should do the same for SAIF simulation? (@imphil: You have deep knowledge of fusesoc! What would you expect here?)

Reply:
I removed because you are right, there would be now way to have a .saif file under build directory before running this one. Thanks for noticing it, line was there because I was first testing the script from an already built project!